### ISSUE TITLE
Disconnecting throws an error

### ISSUE THREAD

**Issue by Rickyling97** [navett52/Argonauts#4]
7:47] [Server thread/ERROR]: [Argonauts] Unable to retreive connection
java.sql.SQLException: ResultSet closed
	at org.sqlite.core.CoreResultSet.checkOpen(CoreResultSet.java:69) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.core.CoreResultSet.markCol(CoreResultSet.java:96) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3ResultSet.getDouble(JDBC3ResultSet.java:360) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.initAdventurer(ArgonautsDatabase.java:90) [Argonauts.jar:?]
	at com.phenom.argonauts.listeners.JoinListener.onAdventurerJoin(JoinListener.java:22) [Argonauts.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:499) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:484) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.PlayerList.onPlayerJoin(PlayerList.java:346) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.PlayerList.a(PlayerList.java:166) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.LoginListener.b(LoginListener.java:159) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.LoginListener.e(LoginListener.java:57) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.NetworkManager.a(NetworkManager.java:233) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.ServerConnection.c(ServerConnection.java:140) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.MinecraftServer.D(MinecraftServer.java:842) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.DedicatedServer.D(DedicatedServer.java:405) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.MinecraftServer.C(MinecraftServer.java:678) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at net.minecraft.server.v1_12_R1.MinecraftServer.run(MinecraftServer.java:576) [spigot.jar:git-Spigot-99c2a37-d35483b]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]

**Comment by navett52** [comment]
Odd, it's throwing this error because of the join listener, but you say you're disconnecting when this happens?

com.phenom.argonauts.listeners.JoinListener.onAdventurerJoin(JoinListener.java:22)

Intriguing

**Comment by navett52** [comment]
Will look into this. Logon servers are down so I am currently unable to look into this further

**Comment by navett52** [comment]
I can't replicate the error. It might be because I'm testing on 1.11.2, but I wouldn't think that would be an issue. I'll update and try it with 1.12 and maybe see if that causes an error.

**Comment by BerndiVader** [comment]
Thats caused because response is empty. Maybe the db doesnt contain the player or its an issue in the sql string in ArgonautsDatabase.java @ line 74.

**Comment by navett52** [comment]
Yeah, what I'm not understanding though is why it's throwing an error from the join listener when he's disconnecting

**Comment by BerndiVader** [comment]
No thats a mistake of ricky, he get this on join.

Edit: btw the thing with the connection closed... you have 2 options: first one is: make the connection class static the other do a timer that keep the conn alive..... i know pretty ugly but afaik one of the save ways in java.

**Comment by navett52** [comment]
It appears my query is broken somewhere. After running some tests the Adventurer record on join is posting fine, but when the player gets back on and needs to be initialized something in my query is wrong causing the ResultSet to be empty. I will work on it

**Comment by navett52** [comment]
Fixed it. There were a few issues. When I was testing I added in some fake data to make sure my queries were working. So when I tried to use my query to grab attack values that didn't exist yet the DB couldn't find it. I worked around that by adding the default Adventurer Role and Style. As soon as a player joins the server for the first time they will be the Adventurer style under the Adventurer role. During this time I believe they should have very basic abilities like 'return'. So as it stands, the add adventurer function, initialize adventurer function, and save adventurer function all should work.
